cpp/ql/src/semmle/code/cpp/ir/implementation/Opcode.qll

```diff
@@ -211,10 +216,10 @@ module Opcode {
   class CallSideEffect extends MayWriteSideEffectOpcode, TCallSideEffect { override final string toString() { result = "CallSideEffect" } }
   class CallReadSideEffect extends ReadSideEffectOpcode, TCallReadSideEffect { override final string toString() { result = "CallReadSideEffect" } }
   class IndirectReadSideEffect extends ReadSideEffectOpcode, MemoryAccessOpcode, TIndirectReadSideEffect { override final string toString() { result = "IndirectReadSideEffect" } }
-  class IndirectWriteSideEffect extends WriteSideEffectOpcode, MemoryAccessOpcode, TIndirectWriteSideEffect { override final string toString() { result = "IndirectWriteSideEffect" } }
+  class IndirectMustWriteSideEffect extends MustWriteSideEffectOpcode, MemoryAccessOpcode, TIndirectWriteSideEffect { override final string toString() { result = "IndirectWriteSideEffect" } }
```

Comment:
I can't find an example of this opcode being used in the `raw_ir.expected` output. Is it easy enough to add one?

Reply:
We'd need a function model that specifies it, which means either finding a standard library function with a guaranteed single-element write or using a test function and changing all the tests from `.qlref`s to `.ql`s

Reply:
How about [`fopen_s`](https://en.cppreference.com/w/c/io/fopen)?

Reply:
I found an example now. I don't know why I didn't see it before. It's the `memcpy` on line 1155 of ir.cpp. So there's no need for you to model `fopen_s`.